src/mac/main.mm

```diff
@@ -137,12 +137,31 @@ Controller controller(std::make_unique<CommandWithArguments>(commandNameToComman
 
 @end
 
+class SafariApplication final : public Application
+{
+public:
+    using Application::Application;
+
+    bool event(QEvent *e) final {
+        if (e->type() == QEvent::User) {
```

Comment:
Qt recommends to register and reserve a custom event type with [`registerEventType()`](https://doc.qt.io/qt-5/qevent.html#Type-enum) to avoid custom event collisions:

> _For convenience, you can use the registerEventType() function to register and reserve a custom event type for your application. Doing so will allow you to avoid accidentally re-using a custom event type already in use elsewhere in your application._

But we might get rid of the custom event altogether, see below.

Reply:
Its optional when we use more custom events. There is only one